### PR TITLE
#305 add number sign before each number in lessons

### LINKED ISSUE #305: Put a number sign in front of each number or digit
In lessons, some numbers and digits are not preceded by a number sign. This was done on purpose, but Viktoria Nikolaevna pointed out that we should put a number sign *always* before a number (otherwise students may get used to writing numbers without a number sign, she says).

In practice, numbers and digits are never preceded by a number sign.

So:

- [x] Put a number sign everywhere in theory
- [x] Ask V.N.: what to do with practice? It would be tedious to always write it before each digit. Maybe just leave it as is?

### PR DESCRIPTION
**Pull request by zuevval**
Closes #305

### CODE REVIEW

**Comment by zuevval** [comment]
I will ask VN and check the changed course, then mark this request as ready

File: app/src/main/java/com/github/braillesystems/learnbraille/res/GolubinaCourse.kt [hunk 1]
```diff
@@ -230,9 +230,14 @@ internal val golubinaIntroLessons by lessons {
...
                     Например, цифра 3, как и число 3 - это цифровой знак + Ц.
                     Число двадцать четыре - это цифровой знак, затем буквы Б и Д. 
-                    В уроках мы для краткости не будем всякий раз ставить цифровой знак."""
+                    В следующих шагах просмотрите, а затем введите число «двенадцать тысяч
+                    триста сорок пять»."""
         )
-        showAndInputChars("12345")
+        +Show(content.markers.getValue(MarkerType.NumberSign))
+        for (char in "12345") {
+            +Input(content.symbols.getValue(char))
```
Comment (winter-yuki) [inline app/src/main/java/com/github/braillesystems/learnbraille/res/GolubinaCourse.kt:238]: `Show` should be here

File: app/src/main/java/com/github/braillesystems/learnbraille/res/_DSL.kt [hunk 2]
```diff
@@ -22,3 +22,8 @@ fun StepsBuilder.slateStylusLine(char: Char) {
         "Запишите на брайлевском приборе строку, состоящую из одного символа: $char."
     ).annotate(StepAnnotation.slateStylusRequired)
 }
+
+fun StepsBuilder.inputNumber(number: Int) {
+    +Input(content.markers.getValue(MarkerType.NumberSign))
```
Comment (winter-yuki) [inline app/src/main/java/com/github/braillesystems/learnbraille/res/_DSL.kt:27]: maybe annotate this step and make it possible to disable it in the settings?

**Review by winter-yuki — CHANGES_REQUESTED** [review]
Found a mistake

File: app/src/main/java/com/github/braillesystems/learnbraille/res/_DSL.kt (hunk 2 above)
Comment (zuevval) [reply app/src/main/java/com/github/braillesystems/learnbraille/res/_DSL.kt:27]: Viktoria Nikolaevna insists that we should teach to put number signs everywhere

File: app/src/main/java/com/github/braillesystems/learnbraille/res/GolubinaCourse.kt (hunk 1 above)
Comment (zuevval) [reply app/src/main/java/com/github/braillesystems/learnbraille/res/GolubinaCourse.kt]: many thanks! Fixed it. Will check everything one more time before marking this request as ready

File: app/src/main/java/com/github/braillesystems/learnbraille/res/_DSL.kt (hunk 2 above)
Comment (winter-yuki) [reply app/src/main/java/com/github/braillesystems/learnbraille/res/_DSL.kt:34]: We have a special chat to validate VN proposals

Comment (zuevval) [reply app/src/main/java/com/github/braillesystems/learnbraille/res/_DSL.kt:34]: We shall discuss it tomorrow in the call

Comment (zuevval) [reply app/src/main/java/com/github/braillesystems/learnbraille/res/_DSL.kt:34]: decided not to do

**Comment by zuevval** [comment]
VN suggested to mention somewhere that in practice we do not put number signs for technical reasons. I would like to mention this in lessons and in the practice help.

File: app/src/main/java/com/github/braillesystems/learnbraille/res/GolubinaCourse.kt [hunk 3]
```diff
@@ -230,9 +230,14 @@ internal val golubinaIntroLessons by lessons {
                     Они получаются из букв А, Б, Ц, Д, Е добавлением цифрового знака.
                     Например, цифра 3, как и число 3 - это цифровой знак + Ц.
                     Число двадцать четыре - это цифровой знак, затем буквы Б и Д. 
-                    В уроках мы для краткости не будем всякий раз ставить цифровой знак."""
+                    В следующих шагах просмотрите, а затем введите число «двенадцать тысяч
+                    триста сорок пять»."""
         )
-        showAndInputChars("12345")
+        +Show(content.markers.getValue(MarkerType.NumberSign))
```
Comment (winter-yuki) [inline app/src/main/java/com/github/braillesystems/learnbraille/res/GolubinaCourse.kt:236]: It is more consistent to abstract it away too.

**Review by winter-yuki — APPROVED** [review]
I'd like to annotate all the number signs and provide the possibility to hide them

File: app/src/main/java/com/github/braillesystems/learnbraille/res/GolubinaCourse.kt (hunk 3 above)
Comment (zuevval) [reply app/src/main/java/com/github/braillesystems/learnbraille/res/GolubinaCourse.kt:236]: Well, it's not the only place where a "bare" number sign is left in the course. So I suppose it is no worse than any other ShowMarker step.

Comment (zuevval) [reply app/src/main/java/com/github/braillesystems/learnbraille/res/GolubinaCourse.kt:236]: But I've added this to #257

**Comment by zuevval** [comment]
> I'd like to annotate all the number signs and provide the possibility to hide them

VN would mind. She is for the compulsion to put number signs. Logged your proposal in #257

**Pull request merged**